apps/api/R/submit.workflow.R

```diff
@@ -48,23 +48,35 @@ submit.workflow.list <- function(workflowList, userDetails, dbcon = global_db_po
     )
   )
 
-  if (!is.null(workflowList$model$id) &&
-        (is.null(workflowList$model$type) || is.null(workflowList$model$revision))) {
-    res <- dplyr::tbl(dbcon, "models") %>%
-      select(id, model_name, revision) %>%
-      filter(id == !!workflowList$model$id) %>%
-      collect()
-
-    workflowList$model$type <- res$model_name
-    workflowList$model$revision <- res$revision
+  if (is.null(workflowList$model$id)) {
+    stop("Must provide model ID.")
```

Comment:
one option to make it nicer is to use:
```
function(res){
  msg <- "Your request did not include a required parameter."
  res$status <- 400 # Bad request
  return(list(error=jsonlite::unbox(msg)))
```
this will return a 400 message, I think stop will result in a 500

Reply:
Great idea, thanks!

Reply:
Addressed in https://github.com/PecanProject/pecan/pull/2728/commits/453e9a3abf7d03ba14239b4eba71fea178ff2b1b.

Reply:
Actually, scratch that. I'm going to respectfully punt on this because it's a bit more work than I expected (because `POST workflow` calls `submitWorkflow`, which in turn calls several other functions, and the `res` has to be propagated through all of them.

Reply:
Actually, double scratch that. See 35ef735f1.

Reply:
Also, for the above fix to work effectively, you'll need https://github.com/PecanProject/rpecanapi/pull/8.